### PR TITLE
Fix annotation for launcher property

### LINKED ISSUE #15364: "org.gradle.jvm.toolchain.internal.DefaultToolchainJavaLauncher cannot be serialized" error
Follow the https://docs.gradle.org/6.8-rc-1/userguide/toolchains.html#sec:plugins example and add any @OutputFile property (with convention) and observe the "`org.gradle.jvm.toolchain.internal.DefaultToolchainJavaLauncher` cannot be serialized" error. 

This effectively makes the use of the `javaToolchains` impossible with any real world task having any outputs.

```kotlin
import org.gradle.api.DefaultTask
import org.gradle.api.plugins.JavaPluginExtension
import org.gradle.api.tasks.Input
import org.gradle.api.tasks.OutputFile
import org.gradle.api.tasks.TaskAction
import org.gradle.jvm.toolchain.JavaLauncher
import org.gradle.jvm.toolchain.JavaToolchainService
import org.gradle.kotlin.dsl.getByType
import org.gradle.kotlin.dsl.property
import javax.inject.Inject


// FROM GRADLE EXAMPLE
open class CustomTaskUsingToolchains : DefaultTask {

    @get:Input
    val launcher = project.objects.property<JavaLauncher>()

    @OutputFile
    val output = project.objects.fileProperty().convention(project.layout.buildDirectory.file("hello.txt"))

    @Inject
    constructor() {
        // Access the default toolchain
        val toolchain = project.extensions.getByType<JavaPluginExtension>().toolchain

        // acquire a provider that returns the launcher for the toolchain
        val service = project.extensions.getByType<JavaToolchainService>()
        val defaultLauncher = service.launcherFor(toolchain)

        // use it as our default for the property
        launcher.convention(defaultLauncher)
    }

    @TaskAction
    fun showConfiguredToolchain() {
        println(launcher.get().executablePath)
        println(launcher.get().metadata.installationPath)
        output.get().asFile.writeText("Hello World!")
    }
}

tasks.register<CustomTaskUsingToolchains>("showDefaultToolchain")

tasks.register<CustomTaskUsingToolchains>("showCustomToolchain") {
    launcher.set(javaToolchains.launcherFor {
        languageVersion.set(JavaLanguageVersion.of(8))
    })
}

```

Environment: MacOS 11, IntelliJ Ultimate 2020.2.4, Gradle 6.8-RC1, Kotlin 1.4.20

A possible workaround for now is to use the bare bone, no outputs, toolchain task as in the example and then depend on it.
However, this is cumbersome and not ideal!

### PR DESCRIPTION
**Pull request by bmuskalla**
Fixes #15364

### CODE REVIEW

**Comment by bmuskalla** [comment]
@bot-gradle test this

File: subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle [hunk 1]
```diff
@@ -2,16 +2,16 @@ import javax.inject.Inject;
 
 class CustomTaskUsingToolchains extends DefaultTask {
 
-    @Input
-    final Property<JavaLauncher> launcher = project.objects.property(JavaLauncher.class);
+    @Nested
+    final Property<JavaLauncher> launcher = project.objects.property(JavaLauncher.class)
```
Comment (wolfs) [inline subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle:6]: Is there any reason we don't use abstract getters in the sample?

Comment (bmuskalla) [reply subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle:6]: Not that I'm aware of. Would that be our general recommendation? I personally find dealing with abstract methods/abstract class for the sake of decoration/injection quite confusing for a concrete implementation.

Comment (wolfs) [reply subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle:6]: That is at least my recommendation. I think it is much better to declare an abstract getter than accessing the project. You would end up with something like this:
```
@Nested
Property<JavaLauncher> getLauncher()
```

You can still access the property in Groovy as you did before.

**Review by ljacomet — APPROVED** [review]
LGTM!
I would do the change suggested by @wolfs

File: subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle (hunk 1 above)
Comment (bmuskalla) [reply subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle]: Fair enough, fixed

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Pull request merged**